pkg/controller/datavolume/snapshot-clone-controller.go

```diff
@@ -376,6 +422,15 @@ func getTempHostAssistedSourcePvcName(dv *cdiv1.DataVolume) string {
 
 func (r *SnapshotCloneReconciler) cleanup(syncState *dvSyncState) error {
 	dv := syncState.dvMutated
+	if err := r.populateSourceIfSourceRef(dv); err != nil {
+		return err
+	}
+
+	if dv.DeletionTimestamp != nil {
+		if err := r.reconcileVolumeCloneSourceCR(syncState); err != nil {
```

Comment:
sure we dont want to call this also when dv succeeded?

Reply:
It will get called here:  https://github.com/kubevirt/containerized-data-importer/blob/9da9c01e032c1c6d26e242f3b3fe5789c456eb6e/pkg/controller/datavolume/snapshot-clone-controller.go#L233

Reply:
if some reason we will exist here: https://github.com/kubevirt/containerized-data-importer/blob/9da9c01e032c1c6d26e242f3b3fe5789c456eb6e/pkg/controller/datavolume/snapshot-clone-controller.go#L181 or here: https://github.com/kubevirt/containerized-data-importer/blob/9da9c01e032c1c6d26e242f3b3fe5789c456eb6e/pkg/controller/datavolume/snapshot-clone-controller.go#L193
then we will never delete the volumeclonesource. dont see why should just call it few lines after https://github.com/kubevirt/containerized-data-importer/pull/2750/files/ac7a342dcd831bb0746e634be774942d05d20086..9da9c01e032c1c6d26e242f3b3fe5789c456eb6e#diff-57c4756c3e19744255585cd838f48d98a4410e30b31813b74b23874c998b3268R441
after the check of deleted and succeeded..

Reply:
Unless there is an unexpected error (which is retried), it should get called.  But I suppose the behavior of the base controller could change and mess things up as well.  This is such spaghetti code.  IMO controllers should always execute the same code block at each reconcile.  I don't like that the derived controller `sync` is not called if the DataVolume is deleted.

Anyway, my intention was to reduce special cases and have the cleanup function do less.  Because I don't think it should exist in the first place.